src/operator/nn/pooling-inl.h

```diff
@@ -53,6 +53,7 @@ struct PoolingParam : public dmlc::Parameter<PoolingParam> {
   bool cudnn_off;
   dmlc::optional<int> p_value;
   dmlc::optional<bool> count_include_pad;
+  dmlc::optional<int> layout;
```

Comment:
Why only adding this option to pooling?

Reply:
Convolutions already have it (layout parameter), batchnorm has notion of axis as well.